forms-flow-web/src/components/ServiceFlow/list/sort/CreateNewFilter.js

```diff
@@ -13,36 +14,59 @@ export default function CreateNewFilterDrawer() {
   const [inputVisibility, setInputVisibility] = useState({});
   const [definitionKeyId, setDefinitionKeyId] = useState("");
   const [candidateGroup, setCandidateGroup] = useState([]);
-  const [taskName, setTaskName] = useState("");
+  const [assignee, setAssignee] = useState("");
   const [includeAssignedTasks, setIncludeAssignedTasks] = useState(false);
-  const defaultRow = { id: "", name: "", label: "", add: true, delete: false };
-  const [rows, setRows] = useState([defaultRow]);
-  const [newRow, setNewRow] = useState([]);
   const [permissions, setPermissions] = useState("");
   const [identifierId, setIdentifierId] = useState("");
   const [selectUserGroupIcon, setSelectUserGroupIcon] = useState("");
   const [specificUserGroup, setSpecificUserGroup] = useState("");
-  const userName = useSelector((state) => state.user?.userDetail?.name);
+  const userName = useSelector((state) => state.user?.userDetail?.preferred_username);
+  const [variables,setVariables] = useState([]);
+  const [inputValues, setInputValues] = useState([{ name: '', label: '' }]);
+
+  useEffect(()=>{
+    setVariables(() => {
+      if (inputValues.length === 1 && inputValues[0].name === '' && inputValues[0].label === '') {
+        return [];
+      } else {
+        return inputValues.map((row) => ({
```

Comment:
why need to do here map ? could you please explain

Reply:
here map is used to set all input values to variable state